tests/test_security_curve.cpp

```diff
@@ -40,6 +40,10 @@
 #  include <unistd.h>
 #endif
 
+#include "../src/tweetnacl.h"
```

Comment:
I'm afraid that would be worse - it would break all applications that do static builds since there's now an extra library, and would make all build systems even more complicated.

It's already broken now - for example, it's no longer possible to build with libunwind:

```
  CXXLD    tests/test_security_curve
/usr/bin/ld: src/err.o: undefined reference to symbol '_Ux86_64_getcontext'
//usr/lib/x86_64-linux-gnu/libunwind.so.8: error adding symbols: DSO missing from command line
```

Likely some flags missing somewhere.